### ISSUE TITLE
Create descriptive types for RPC call arguments and return types.

### ISSUE THREAD

**Issue by aupiff** [airalab/hs-web3#15]
Already, `Address` serves as a very descriptive type in the `web3` library. I propose we add additional types such as `UInt256`, `Bytes32` etc. I list three versions of `eth_getBalance` below, each more descriptive than the last.

CURRENT: `eth_getBalance :: Provider a => Address -> CallMode -> Web3 a Text`
OK: `eth_getBalance :: Provider a => Address -> CallMode -> Web3 a Integer`
GOOD: `eth_getBalance :: Provider a => Address -> CallMode -> Web3 a UInt256`

**Comment by akru** [comment]
You are absolutely right, it grows readability and safety.

But, for example, `Bytes32` already implemented in https://github.com/airalab/hs-web3/blob/master/src/Network/Ethereum/Web3/Encoding/Bytes.hs as `BytesN 32`. It can be potentially described as type synonym

```haskell
type Bytes32 = BytesN 32
```

So, can we prepare table with EVM types and its synonyms when it already exists?

For example:

| EVM | Haskell | hs-web3 |
|---------------|---------------------|-----------------|
| `UInt8` | `Word8` | `type UInt8 = Word8` |
| `Bytes32` | `Bytes` | `BytesN 32` |

**Comment by akru** [comment]
@aupiff, do you make PR with this enchancement?
I want to make release 0.6 of library at next week, do it in this release or not?

**Comment by aupiff** [comment]
@akru Yep, I'll submit a PR for this within the next couple of days so it can be included in your next release.

**Comment by aupiff** [comment]
From web3 JSONRPC docs:
```
A note on specifying topic filters:

Topics are order-dependent. A transaction with a log with topics [A, B] will be matched by the following topic filters:

[] "anything"
[A] "A in first position (and anything after)"
[null, B] "anything in first position AND B in second position (and anything after)"
[A, B] "A in first position AND B in second position (and anything after)"
[[A, B], [A, B]] "(A OR B) in first position AND (A OR B) in second position (and anything after)"
```
Given the above explanation, I think its necessary to change the `Filter` type to have a field `filterTopics :: !(Maybe [Maybe [Text]])` instead of `filterTopics :: !(Maybe [Maybe Text])`.

```
-- | Low-level event filter data structure
data Filter = Filter
  { filterAddress   :: !(Maybe Address)
  , filterTopics    :: !(Maybe [Maybe Text])
  , filterFromBlock :: !(Maybe Text)
  , filterToBlock   :: !(Maybe Text)
  } deriving Show
```

**Comment by akru** [comment]
In #34 added `IntN`, `UIntN` and sized vector types.
